src/libs/antares/exception/LoadingError.hpp

```diff
@@ -0,0 +1,148 @@
+/*
+** Copyright 2007-2018 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#pragma once
+
+#include <stdexcept>
+#include <yuni/yuni.h>
+#include <yuni/string.h>
+#include <antares/study/fwd.h>
+
+namespace Antares
+{
+namespace Error
+{
+class StudyFolderDoesNotExist : public std::runtime_error
+{
+public:
+    explicit StudyFolderDoesNotExist(const Yuni::String& folder);
+};
+
+class ReadingStudy : public std::runtime_error
+{
+public:
+    explicit ReadingStudy();
```

Comment:
This constructor has no parameter. Why the **explicit** keyword then ?
The question stands for any other constructor that has no parameter.

Reply:
I placed it here by habit, but it's not really useful in this case.